analytic_tag_dimension/tests/test_analytic_dimension.py

```diff
@@ -194,4 +198,78 @@ def test_zz_dimension_rename(self):
         # as not everything is undone in this renaming
         self.dimension_1.write({"code": "test_renamed"})
         self.assertIn("x_dimension_test_renamed", self.analytic_line_obj._fields)
-        self.assertIn("x_dimension_test_renamed", self.env["account.move.line"]._fields)
+        self.assertIn("x_dimension_test_renamed", self.account_move_line_obj._fields)
+
+    def test_zz_dimension_rename2(self):
+        # It should executed the last one for avoiding side effects
+        # as not everything is undone in this renaming
+        self.dimension_1.write({"name": "test_renamed"})
+        self.assertIn("x_dimension_test_renamed", self.analytic_line_obj._fields)
+        self.assertIn("x_dimension_test_renamed", self.account_move_line_obj._fields)
+
+    def assertInvoiceReportValues(self, expected_values_list):
+        reports = self.env["account.invoice.report"].search(
+            [], order="price_subtotal DESC, quantity ASC"
+        )
+        expected_values_dict = [
+            {
+                "price_average": vals[0],
+                "price_subtotal": vals[1],
+                "quantity": vals[2],
+            }
+            for vals in expected_values_list
+        ]
+        self.assertRecordValues(reports, expected_values_dict)
+
+    def assertInvoiceReportValues_update_custom_fields(self, expected_values_list):
+        reports = (
+            self.env["account.invoice.report"]
+            .with_context(update_custom_fields=True)
+            .search([], order="price_subtotal DESC, quantity ASC")
+        )
+        expected_values_dict = [
+            {
+                "price_average": vals[0],
+                "price_subtotal": vals[1],
+                "quantity": vals[2],
+            }
+            for vals in expected_values_list
+        ]
+        self.assertRecordValues(reports, expected_values_dict)
+
+    def test_invoice_report_multiple_types(self):
+        self.assertInvoiceReportValues(
+            [
+                [1500.0, 30000.0, 20.0],
```

Comment:
This result is so dependent on the existing data that will fail when other module introduces new data. Please always assert specific and controlled data.